mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -484,8 +480,9 @@ class Word2VecModel private[spark] (
    * @return vector representation of word
    */
   def transform(word: String): Vector = {
-    model.get(word) match {
-      case Some(vec) =>
+    wordIndex.get(word) match {
+      case Some(ind) =>
+        val vec = wordVectors.slice(ind * vectorSize, ind * vectorSize + vectorSize)
```

Comment:
Are you sure? I think a copy will be produced anyway. It seems if it is a collection.view then it does not produce a copy of collection.

Ref: (http://stackoverflow.com/a/6799739/1170730)